library/alloc/src/collections/vec_deque/mod.rs

```diff
@@ -2270,9 +1981,9 @@ impl<T, A: Allocator> VecDeque<T, A> {
         // Extend or possibly remove this assertion when valid use-cases for growing the
         // buffer without it being full emerge
         debug_assert!(self.is_full());
-        let old_cap = self.cap();
-        self.buf.reserve_exact(old_cap, old_cap);
-        assert!(self.cap() == old_cap * 2);
+        let old_cap = self.capacity();
+        // if the cap was 0, we need to reserve at least 1.
+        self.buf.reserve(old_cap, old_cap.max(1));
```

Comment:
The comment on the method says "double", but I'm pretty sure this implementation no longer cares that it's precisely a doubling.

Thus I think this is just the "grow an appropriate amount to avoid quadratic behaviour" method now, which implies that it probably wants to use https://stdrs.dev/nightly/x86_64-unknown-linux-gnu/alloc/raw_vec/struct.RawVec.html#method.reserve_for_push instead, like
```suggestion
        self.buf.reserve_for_push(old_cap);
```
as that's an optimized version of `self.reserve(old_cap, 1);` and has the amortization logic.